golang-sample/cloudbuild.yaml

```diff
@@ -0,0 +1,61 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+steps:
+  # Initialize the module
+  - name: golang
+    args: ['go', 'mod', 'init', 'github.com/GoogleCloudPlatform/cloud-build-samples/golang']
+
+  # Clean up dependencies
+  - name: golang
+    args: ['go', 'mod', 'tidy']
```

Comment:
blocker: My view is this should not be done, as it's rewriting code/dependency manifests as part of the build, which means the build is less deterministic as a result.

considerations:

This might make sense as a static analysis test with the new `-e` flag in go 1.16, which would fail the build if `go mod tidy` sees a conflict that needs fixing.

If you want to include go static analysis checks like that, it's good practice to use `go vet` and a `go fmt` check is also common.